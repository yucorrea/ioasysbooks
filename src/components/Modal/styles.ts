import { Animated, Dimensions } from 'react-native';
import styled from 'styled-components/native';

const { width, height } = Dimensions.get('window');

export const ModalContainer = styled(Animated.View)`
  width: ${width}px;
  height: ${height}px;
  padding: 16px;

  position: absolute;
  z-index: 2;

  justify-content: center;
  align-items: center;

  background: rgba(0, 0, 0, 0.5);
`;

export const ModalContent = styled(Animated.View)`
  width: 100%;
  border-radius: 4px;
  background: ${({ theme }) => theme.colors.background};
  padding: 16px;
`;

export const Header = styled.View`
  width: 100%;
  align-items: flex-end;
`;
