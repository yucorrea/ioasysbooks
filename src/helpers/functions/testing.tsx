import React from 'react';
import renderer from 'react-test-renderer';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';

import theme from '../../config/theme';

export const matchSnapshotWithTheme = (component :  React.ReactElement) => {
  const tree = renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>).toJSON();
  expect(tree).toMatchSnapshot();
};

export const renderWithTheme = (component : React.ReactElement, options?: RenderOptions) => {
  return render(<ThemeProvider theme={theme}> {component} </ThemeProvider>, options);
};
