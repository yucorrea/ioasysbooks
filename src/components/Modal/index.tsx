import React, { useState } from 'react';
import { Animated, Modal } from 'react-native';
import { IconButton } from '../IconButton';

import { ModalContainer, ModalContent, Header } from './styles';

interface Props {
  show: boolean;
  close: () => void;
  children: React.ReactNode;
}

export function ModalComponent({ show, close, children }: Props) {

  return (
    <Modal visible={show} onRequestClose={close} animationType="fade" transparent>
      <ModalContainer>
        <ModalContent>
          <Header>
            <IconButton onPress={close} icon="close-outline" />
          </Header>
          {children}
        </ModalContent>
      </ModalContainer>
    </Modal>
  );
}
