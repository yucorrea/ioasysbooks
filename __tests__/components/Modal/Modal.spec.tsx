import React from 'react';
import { View, Text } from 'react-native';

import Modal from '@components/Modal';

import { matchSnapshotWithTheme, renderWithTheme } from '@helpers/functions/testing';

const Children = () => {
  return (
    <View testID='children-component'>
      <Text>Modal</Text>
    </View>
  );
};

describe('Modal Component', () => {
  it('matches snapshots', () => {
    const mockFunction = jest.fn();
    const snap = matchSnapshotWithTheme(
      <Modal show close={mockFunction}>
        <Children />
      </Modal>,
    );
    expect(snap).toMatchSnapshot();
  });

  it('render correctly', () => {
    const mockFunction = jest.fn();
    renderWithTheme(
      <Modal show close={mockFunction}>
        <Children />
      </Modal>,
    );
  });


  it('check if contain children in modal ', () => {
    const mockFunction = jest.fn();
    const { getByText } = renderWithTheme(
      <Modal  show close={mockFunction}>
        <Children />
      </Modal>,
    );

    expect(getByText('Modal')).toBeTruthy();
  });

  it('when pass props falsy modal hide', () => {
    const mockFunction = jest.fn();
    const { getByTestId } = renderWithTheme(
      <Modal testID="modal-component" show={false} close={mockFunction}>
        <Children />
      </Modal>,
    );

    const modal = getByTestId("modal-component");
    expect(modal.props.visible).toBeFalsy();
  });
});
