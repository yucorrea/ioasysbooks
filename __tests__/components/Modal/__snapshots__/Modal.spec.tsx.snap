// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component matches snapshots 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "height": 1334,
          "justifyContent": "center",
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 4,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "width": "100%",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "rgba(51, 51, 51, 0.2)",
              "borderRadius": 16,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 32,
              "justifyContent": "center",
              "opacity": 1,
              "width": 32,
            }
          }
          testID="button-modal-close"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ïˆ§
          </Text>
        </View>
      </View>
      <View
        testID="children-component"
      >
        <Text>
          Modal
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Modal Component matches snapshots 2`] = `undefined`;
